internal/olm/operator/internal/service.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olm
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// withTCPPort returns a function that appends a service port to a Service's
+// port list with name and TCP port portNum.
+func withTCPPort(name string, portNum int32) func(*corev1.Service) {
+	return func(service *corev1.Service) {
```

Comment:
IHMO `withTCPPort` is not helpful for we know what it does. WDYT about something as `buildServiceWithTCPPort`?

Reply:
The comment and return type makes its intended use obvious :smile:.

Reply:
Sorry, if I could not make clear my POV.

By looking a code as for example: `newResource(deployment, service, withTCPPort(pkg))`

I am not sure that my understanding will be that the `withTCPPort` will impl a new Service with a TCP port without the need to go there and check the withTCPPort code implementation. 

However, if it was something as `newService().withTCPPort()` or  `newResource(deployment, service, newTCPService(pkg))` then, I think that is not required check the details of the impl to know what it does. 

My suggestion would be just rename the func to make clear what it does without the need to check its implementation.

Reply:
See my comment above: https://github.com/operator-framework/operator-sdk/pull/2313#discussion_r357336492.

Reply:
The `withTCPPort` makes sense now that I've understood the paradigm being used here. I'm okay with it.